README.md

```diff
@@ -142,16 +142,17 @@ dependencies {
 ```groovy
 // Add Dagger dependencies
 dependencies {
-  compile 'com.google.dagger:dagger:2.x'
+  api 'com.google.dagger:dagger:2.x'
   annotationProcessor 'com.google.dagger:dagger-compiler:2.x'
 }
 ```
+For more information on api vs implementation please see https://developer.android.com/studio/build/gradle-plugin-3-0-0-migration?utm_source=android-studio#new_configurations
```

Comment:
That link seems to be broken - I think this is now the best reference: https://developer.android.com/studio/build/dependencies#dependency_configurations

Interestingly the docs there say about `api`:
> This configuration behaves just like compile (which is now deprecated)

Reply:
Should we instead link to https://docs.gradle.org/current/userguide/java_library_plugin.html#sec:java_library_separation for non-android users too?

Also, let's make give this a name so we don't have a super long URL in the docs.